### PR TITLE
ASoC: SOF: topology: add code to parse config params for ACPDMIC dai

### PR DESCRIPTION
**Pull request by ajitkupandey**
Add sof_ipc_dai_acpdmic_params and tokens to parse dmic channels and
rate params from topology and send dai config ipc for ACPDMIC dai.

Signed-off-by: Ajit Kumar Pandey <AjitKumar.Pandey@amd.com>

### CODE REVIEW

File: include/sound/sof/dai-amd.h [hunk 1]
```diff
@@ -18,4 +18,12 @@ struct sof_ipc_dai_acp_params {
 	uint32_t fsync_rate;    /* FSYNC frequency in Hz */
 	uint32_t tdm_slots;
 } __packed;
+
+/* ACPDMIC Configuration Request - SOF_IPC_DAI_AMD_CONFIG */
+struct sof_ipc_dai_acpdmic_params {
+
```
Comment (ujfalusi) [inline include/sound/sof/dai-amd.h]: Remove the blank line?

You don't need `struct sof_ipc_hdr hdr;` in here?

**Review by ujfalusi — CHANGES_REQUESTED** [review]

File: include/sound/sof/dai-amd.h (hunk 1 above)
Comment (ajitkupandey) [reply include/sound/sof/dai-amd.h]: hdr filed is not used in our firmware code hence not needed

Comment (ranj063) [reply include/sound/sof/dai-amd.h]: lets remove the blank line @ajitkupandey and in the commit message you mention "send the DAI CONFIG IPC" but there's no code to back it up. So, maybe better not to mention it as it is not done during topology parsing

Comment (ajitkupandey) [reply include/sound/sof/dai-amd.h]: Done

**Review by ranj063 — APPROVED** [review]

**Review by bhiregoudar — APPROVED** [review]

**Review by sunilkumardommati — APPROVED** [review]

**Review by ujfalusi — APPROVED** [review]

**Review by plbossart — APPROVED** [review]

**Pull request merged**